packages/flutter/lib/src/widgets/basic.dart

```diff
@@ -2042,6 +2042,7 @@ class NetworkImage extends StatelessWidget {
   @override
   Widget build(BuildContext context) {
     return new RawImageResource(
+      key: new ObjectKey(src),
```

Comment:
The key needs to be based on the ImageResource itself, I think.

I would evaluate the load() before the return statement, then pass in the value to "image" and use something based on that as the key.